File: 10866.py
from collections import deque
import sys

input = sys.stdin.readline

n = int(input())

deque = deque()

for _ in range(n):
    a = input().split()
    if a[0] == 'push_front':
        deque.appendleft(a[1])
    elif a[0] == 'push_back':
        deque.append(a[1])
    elif a[0] == 'pop_front':
        if len(deque) == 0:
            print(-1)
        else:
            print(deque.popleft())
    elif a[0] == 'pop_back':
        if len(deque) == 0:
            print(-1)
        else:
            print(deque.pop())
    elif a[0] == 'size':
        print(len(deque))
    elif a[0] == 'empty':
        if len(deque) == 0:
            print(1)
        else:
            print(0)
    elif a[0] == 'front':
        if len(deque) == 0:
            print(-1)
        else:
            print(deque[0])
    elif a[0] == 'back':
        if len(deque) == 0:
            print(-1)
        else:
            print(deque[-1])

File: c++/10866.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main(){
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);
  
  int n;
  vector<int> v;
  cin >> n;

  for(int i=0; i<n; i++){
    string s;
    cin >> s;
    if(s=="push_front"){
      int a;
      cin >> a;
      v.push_back(a);
      for(int i=v.size()-1; i>0; i--){
          int temp = v[i-1];
          v[i-1] = v[i];
          v[i] = temp;
        }
    } else if(s=="push_back"){
      int a;
      cin >> a;
      v.push_back(a);
    } else if(s=="pop_front"){
      if(v.empty()){
        cout << -1 << '\n';
      }else{
        cout << v.front() << '\n';
        for(int i=0; i<v.size()-1; i++){
          int temp = v[i];
          v[i] = v[i+1];
          v[i+1] = temp;
        }
        v.pop_back();
      }
    } else if(s=="pop_back"){
      if(v.empty()){
        cout << -1 << '\n';
      }else{
        cout << v.back() << '\n';
        v.pop_back();
      }
    } else if(s=="size"){
      cout << v.size() << '\n';
    } else if(s=="empty"){
      cout << v.empty() << '\n';
    } else if(s=="front"){
      if(v.empty()){
        cout << -1 << '\n';
      }else{
        cout << v.front() << '\n';
      }
      
    } else if(s=="back"){
      if(v.empty()){
        cout << -1 << '\n';
      }else{
        cout << v.back() << '\n';
      }
    }
  }

  return 0;
}